stdlib/toolchain/CMakeLists.txt

```diff
@@ -50,6 +50,19 @@ if(SWIFT_BUILD_STDLIB)
   add_subdirectory(CompatibilityBytecodeLayouts)
 endif()
 
+macro(dont_target_arm64e_for_products_configured_in_this_folder)
```

Comment:
Not a huge fan of the macro. `PARENT_SCOPE` may be useful here. Alternatively, depending on how many Darwin SDKs we have, I'd prefer inlining the list remove item in the CMakeLists.txt file for each library. The Compatibility 50 and 51 libraries already have `list(REMOVE_ITEM SWIFT_SDK_WATCHOS_ARCHITECTURES "arm64" "arm64e")` in them. In the event that the list of SDKs changes, we already have to edit the library files anyway to fix the deployment target version.